### DISCUSSION TITLE
Separate repo for themes

### DISCUSSION THREAD

**Discussion by MJECloud** [JanDeDobbeleer/oh-my-posh#611] [Ideas]
Hi, 

this idea just come to my mind. 
Would'nt it be better or simpler to separate OMP themes in their own repo?
Sometimes there is a new version of oh-my-posh but there is only a theme added but nothing changed on the codebase. 

What do you think about it?

**Comment by JanDeDobbeleer** [comment]
Well, both work for me. We can create a public themes repo that is fetched and packed when we push actual changes or we exclude theme additions from creating a release. I'm more in favor of the second option as it keeps things straightforward and achieves the same result. Thoughts?

**Reply by MJECloud** [reply]
Yeah you're right. The second option makes more sense. :)

**Reply by JanDeDobbeleer** [reply]
@MJECloud done

**Reply by lnu** [reply]
I just face the issue now: cinnamon theme updated, but it will be only shipped when some code changes. I understand that shipping a build each time a theme is updated is overkill, but at the same time it does not work as before.
The idea of a public theme repo is nice but means also more work.

**Reply by MJECloud** [reply]
Yeah good point. 
Never thought of it because i've my own theme. 
Maybe it's better to revert the change then?

**Reply by JanDeDobbeleer** [reply]
@lnu fair point.

**Reply by lnu** [reply]
I'm trying something with the github api.
We could retrieve the theme list in the autocomple function(working) and then download the theme when calling set-poshprompt(working too). It needs to be polished a little bit but what do you think about it?

**Reply by lnu** [reply]
Something like this: https://github.com/JanDeDobbeleer/oh-my-posh/pull/681

**Reply by JanDeDobbeleer** [reply]
That's dangerous as compatibility of themes can not be guaranteed with the version of oh-my-posh locally.

**Reply by lnu** [reply]
Yes, but in some ways, it's already the case now. The themes are never synced with the code(they're supposed to work with the latest version). If you don't use the powershell module, you need to download the themes manually and you also can get into trouble in that case. The goal of the PR is to raise all issues around the theme and to find a viable solution(if any).
In the end maybe it's easier to keep releasing a new build when a theme is added/updated?

**Reply by JanDeDobbeleer** [reply]
Yes, that's the only change that makes sense. I try never to break compatibility.
